core/vm/contracts.go

```diff
@@ -158,12 +158,13 @@ func init() {
 
 	// Ensure that this package will panic during init if there is a conflict present with the declared
 	// precompile addresses.
-	for _, k := range precompile.UsedAddresses {
-		if _, ok := PrecompileAllNativeAddresses[k]; ok {
-			panic(fmt.Errorf("precompile address collides with existing native address: %s", k))
+	for _, module := range precompile.RegisteredModules() {
+		address := module.Address()
+		if _, ok := PrecompileAllNativeAddresses[address]; ok {
+			panic(fmt.Errorf("precompile address collides with existing native address: %s", address))
 		}
-		if k == constants.BlackholeAddr {
-			panic(fmt.Errorf("cannot use address %s for stateful precompile - overlaps with blackhole address", k))
+		if address == constants.BlackholeAddr {
+			panic(fmt.Errorf("cannot use address %s for stateful precompile - overlaps with blackhole address", address))
```

Comment:
Should we add a check that none of the registered modules have a conflicting address with each other?

Reply:
we do that here
https://github.com/ava-labs/subnet-evm/blob/6039731200a77a27ea66e872cf86a7e594e403d2/precompile/stateful_precompile_module.go#L70-L74